src/main/resources/db/migration/V2__create_item_cat_auction.sql

```diff
@@ -0,0 +1,43 @@
+BEGIN;
+CREATE TABLE IF NOT EXISTS category (
+  id UUID DEFAULT gen_random_uuid () PRIMARY KEY,
+  name text NOT NULL,
+  subcategory UUID,
+  CONSTRAINT fk_subcategory
+      FOREIGN KEY(subcategory) 
+	  REFERENCES category(id)
+);
+
+CREATE TABLE IF NOT EXISTS item (
+  id UUID DEFAULT gen_random_uuid () PRIMARY KEY,
+  name text NOT NULL,
+  start_price decimal,
+  color text,
+  size numeric,
+  description text NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS auction (
+  id UUID DEFAULT gen_random_uuid () PRIMARY KEY,
+  start_date timestamp,
+  end_date timestamp,
+  highest_bid decimal,
```

Comment:
The auction should have a list of all bids tied to it, not just the value of the highest one. It is a logical requirement (and one we will have in the future I believe) to show a list of all bids to the seller.

Reply:
There is a separate table named "bid" that tracks the records of all bids.
I put this field here to not have to do constant querying of the bid table each time I have to show the highest bid, but rather this field gets updated when the new higher bid is entered and is always ready for display.

Is it okay to have it like that?

Reply:
No, because of a few reasons:

- this field has no constraints set on it. If its existence was necessary, it would need to be coupled with the bid table to ensure its value is valid. As it stands right now, someone can maliciously enter a random value into it and the system has no way to react to that as it's just a random standalone field.
- Unless we are talking about significant performance impacts and performance-critical applications, clarity and logic trump making these shortcuts every time. 
- If your auction has a list of all its bids, you could use an `order by` in the API call to order them by price which would put the highest bid as the first every time without polluting the database. You could also have a separate getter in the auction table which could do the same thing.

Reply:
Okay, I understand. I deleted the column in this PR, and I will implement it in the new way with the bidding table in the bidding_process branch.